esp/src/src-react/index.tsx

```diff
@@ -43,8 +43,8 @@ async function loadUI() {
     const authType = await authTypeResp?.text() ?? "None";
     const userStore = cookieKeyValStore();
     const userSession = await userStore.getAll();
-    if (authType.indexOf("None") < 0 && (userSession["ESPSessionState"] === "false" || userSession["ECLWatchUser"] === "false")) {
-        if (window.location.hash.indexOf("login") < 0) {
+    if (authType.indexOf("None") < 0 && (userSession["ESPAuthenticated"] === undefined || userSession["ESPAuthenticated"] === "false")) {
+        if ([...window.location.hash.matchAll(/login/gi)]?.length <= 0) {
```

Comment:
This should either be:
`[...window.location.hash.matchAll(/login/gi)].length`, or 
`window.location.hash.matchAll(/login/gi)?.length`

The latter being preferable?

Reply:
Should be the first one. matchAll returns a "RegExp String Iterator" object, which doesn't have a length property. I think the "<= 0" part was a mistake, because I had .indexOf there at some point. This will replace the url with /login if it doesn't already contain "login" or "Login" (when length of matches was 0).

Reply:
Agree - the first one should still drop the "?" after the array brackets...